### DISCUSSION TITLE
Linux ANE : add a GtkWidget

### DISCUSSION THREAD

**Discussion by pol2095** [airsdk/Adobe-Runtime-Support#1456] [General]
Hello,

is-it possible to add a GtkWidget other the Air app using an ANE on Linux ?

Thanks

**Comment by ajwfrost** [comment]
Once we release the next version of the SDK with support for #236 then I guess it should be possible to do this .. 
`FREAcquireNativeWindowHandle` will provide the main AIR rendering area as a `GtkWidget` so you should be able to add overlays etc..

**Comment by pol2095** [comment]
It's possible using 
```
GList *window_list = gtk_window_list_toplevels();
g_list_foreach(window_list, (GFunc)g_object_ref, NULL);
GtkWindow* window;
for (GList* i = window_list; i != NULL; i = g_list_next(i)) {
    window = GTK_WINDOW(i->data);
}
```
but when I add a GtkWidget I have an incorrect mouse position in th Air app, really problematic.

an other thing, the first Air nativeWindow seems using 2 gtk window (it's possible to determine the good window)
